Modules/Compatibility/Deprecated/include/itkTreeIteratorBase.hxx

```diff
@@ -124,7 +124,7 @@ TreeIteratorBase<TTreeType>::Add(ValueType element)
 
 template <typename TTreeType>
 bool
-TreeIteratorBase<TTreeType>::Add(int itkNotUsed(childPosition), ValueType element)
+TreeIteratorBase<TTreeType>::Add([[maybe_unused]] int childPosition, ValueType element)
```

Comment:
@hjmjohnson Sorry, I'm not in favor of replacing `itkNotUsed(param)` with `[[maybe_unused]]`.  `itkNotUsed(param)` is stronger than "maybe unused". `itkNotUsed` means _definitely_ not used at all. The compiler ensures that such a parameter is not used.

Reply:
I do like `[[maybe_unused]]` as replacement of `(void)varname`, but not as a replacement for unused parameters. If a parameter is unused, I would either use `itkNotUsed(param)`, or simply leave the parameter unnamed.

Reply:
I recall some places `(void)` was used to ensure that a symbol was not optimized away by the compiler. Does `[[maybe_unused]]` do the same thing?

Reply:
@blowekamp Can you please elaborate? Is the use of `(void)` to prevent optimization compiler-specific? If so, does it still work like that, nowadays? Is there any documentation on this behavior?

Reply:
@blowekamp  I found only one place where this occurs in a test.  It appears that the compiler does preserve the variable when [[maybe_unused]] is called.

Reply:
Thank you for investigating my concern. I was not able to reproduce the scenario with godbolt either.